resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -779,19 +781,19 @@ private[spark] class MesosClusterScheduler(
   }
 
   private def getNewRetryState(
-    retryState: Option[MesosClusterRetryState], status: TaskStatus): MesosClusterRetryState = {
-
-    val (retries, waitTimeSec) = retryState
-      .map { rs => (rs.retries + 1, rs.waitTime) }
-      .getOrElse{ (1, 1) }
-
-    // if a node is draining, the driver should be relaunched without backoff
-    if (isNodeDraining(status)) {
-      new MesosClusterRetryState(status, retries, new Date(), waitTimeSec)
-    } else {
-      val newWaitTime = waitTimeSec * 2
-      val nextRetry = new Date(new Date().getTime + newWaitTime * 1000L)
-      new MesosClusterRetryState(status, retries, nextRetry, newWaitTime)
+      retryState: Option[MesosClusterRetryState], status: TaskStatus): MesosClusterRetryState = {
+    val now = new Date()
+    retryState.map { rs =>
+      val newRetries = rs.retries + 1
+      // if a node is draining, the driver should be relaunched without backoff
+      if (isNodeDraining(status)) {
+        new MesosClusterRetryState(status, newRetries, now, rs.waitTime)
+      } else {
+        new MesosClusterRetryState(status, newRetries, DateUtils.addSeconds(now, rs.waitTime), rs.waitTime * 2)
```

Comment:
one last thing, this line will cause scalastyle check to fail when we upstream this change (line length > 100), this should pass:
```
new MesosClusterRetryState(
          status, newRetries, DateUtils.addSeconds(now, rs.waitTime), rs.waitTime * 2)
```